### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by Nessrine88**
Result before solving n+1 problem
n+1 before
Result after solving n+1 problem
image

I have created integration tests using Capybara to thoroughly test each view in our project. The tests are organized to ensure a logical flow and cover various aspects of the user interface.

**User Index Page:**

1. Verifies the visibility of usernames, profile pictures, and the number of posts for each user.
2. Validates the redirection to a user's show page upon clicking a user.

**User Show Page:**
Ensures the display of the user's profile picture, username, bio, and the first three posts.
Includes a button to view all of a user's posts and verifies the redirection to a user's post index page upon clicking a user's post.
User Post Index Page:

1. Validates the visibility of the user's profile picture, username, post details, comments, likes, and pagination functionality.
2. Verifies the redirection to a post's show page upon clicking a post.

**Post Show Page:**

1. Ensures the visibility of the post's title, author, comments, likes, body, and the details of each comment.
2. These tests provide comprehensive coverage of our application's views and user interactions. Your feedback on this pull request is highly appreciated.

### CODE REVIEW

File: spec/system/posts_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    posts.each do |post|
+      expect(page).to have_content(post.likes.count)
+    end
+  end
+
+  it 'it redirects me to that post\'s show page.When I click post' do
+    visit user_posts_path(user)
+    click_on posts.first.title
+    expect(page).to have_current_path(user_post_path(user, posts.first))
+  end
```
Comment (OyePriscilla) [inline spec/system/posts_index_spec.rb]: - [ ] Please kindly add the missing test highlighted below according to the project requirement. 

Olufe 2

File: spec/system/posts_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  it 'displays the post title' do
+    visit user_post_path(user, post)
+    expect(page).to have_content(post.title)
+  end
+
+  it 'displays the number of comments' do
+    visit user_post_path(user, post)
+    expect(page).to have_content(post.comments.count)
+  end
```
Comment (OyePriscilla) [inline spec/system/posts_show_spec.rb]: - [ ] Please kindly add the missing test highlighted below according to the project requirement. 

Olufe 2

File: spec/system/posts_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+RSpec.configure do |config|
+  config.include FactoryBot::Syntax::Methods
+end
+
+RSpec.describe 'Test the User show page', type: :system do
+  let(:user) { create(:user) }
+  let!(:posts) { create_list(:post, 4, author: user) }
+  let!(:comments) { create_list(:comment, 6, post: posts.first, user:) }
+  let!(:likes) { create_list(:like, 6, post: posts.first, user:) }
```
Comment (OyePriscilla) [inline spec/system/posts_index_spec.rb:12]: - [ ] I was unable to access run your tests successfully untill I added this line of code 👉  `//= link application.js`
 to ` /app/assets/config/manifest.js` file.

**Kindly discard this comment if you did not experience such, perhaps it has to do with my PC environment** 🙏

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Nessrine88 👍 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Write integration tests ✔️
Secure app from n+1 problems. ✔️

## Required Changes ♻️

Please kindly fix the twitter link that is broken in the author section of your README file, I got account suspended message. Kindly remove it if there is no twitter working link at the moment 🙏

_Check the comments under the review also._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zeff96 — APPROVED** [review]
### Status: APPROVED :green_circle: 
Hi @Nessrine88 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

congrats

## Highlights
:heavy_check_mark: linters passing
:heavy_check_mark: All test cases passing
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**